src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("first_login");
+            defaultUser.setAge(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.age = null");
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setPassword(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.password = null");
+    }
+
+    @Test
+    void register_loginAlreadyInStorage_notOk() {
+        User storageUser = new User();
+        storageUser.setLogin("second_login");
+        storageDao.add(storageUser);
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("second_login");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, method allows user duplicates");
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "when login field empty an exception should be thrown");
+    }
+
+    @Test
+    void register_default_ok() {
+        registrationService.register(defaultUser);
+        assertEquals(storageDao.get(defaultUser.getLogin()).getLogin(), defaultUser.getLogin());
+        assertEquals(storageDao.get(defaultUser.getLogin()).getPassword(),
+                defaultUser.getPassword());
+        assertEquals(storageDao.get(defaultUser.getLogin()).getAge(), defaultUser.getAge());
+    }
+
+    @Test
+    void register_passwordLengthLessThan6_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("third_login");
+            defaultUser.setPassword("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "if password length less than 6 an exception should be thrown");
+    }
+
+    @Test
+    void register_AgeLessThan18_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("fourth_login");
+            defaultUser.setAge(16);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, if age < 18 method you should thrown an exception");
+    }
+
+    @Test
+    void register_AgeEquals18_ok() {
+        defaultUser.setAge(18);
+        defaultUser.setLogin("login_fifth");
+        registrationService.register(defaultUser);
+        assertEquals(storageDao.get(defaultUser.getLogin()).getAge(),
```

Comment:
we can not use storage dao here, because it is a unit test - testing individual components as much isolated as possible. let's check the returning value of the register method instead, the same for other tests
\+ it has a bad readability